gems/aws-sdk-core/spec/aws/waiters_spec.rb

```diff
@@ -276,6 +276,90 @@ module Waiters
               expect { client.wait_until(:path_matcher) }
                 .to raise_error(Errors::FailureStateError)
             end
+
+            it 'can match objects' do
+              object = {
+                object_member: {
+                  string_member: 'string'
+                },
+                string_member: 'string',
+                number_member: 1,
+                boolean_member: true,
+                array_member: [1, 2, 3],
+                object_array_member: [
+                  {
+                    object_member: {
+                      string_member: 'string'
+                    },
+                    string_member: 'string'
+                  }
+                ]
+              }
```

Comment:
Nit: could reduce the number of lines here like so - I think. If you do decide to reformat, you might as well do it for others as well

```suggestion
              object = {
                object_member: { string_member: 'string' },
                string_member: 'string',
                number_member: 1,
                boolean_member: true,
                array_member: [1, 2, 3],
                object_array_member: [
                  {
                    object_member: { string_member: 'string' },
                    string_member: 'string'
                  }
                ]
              }
```